src/uct/ib/rc/base/rc_iface.c

```diff
@@ -685,7 +697,8 @@ static UCS_CLASS_CLEANUP_FUNC(uct_rc_iface_t)
         ucs_free(self->eps[i]);
     }
 
-    if (!ucs_list_is_empty(&self->ep_list)) {
+    if (!ucs_list_is_empty(&self->ep_list) ||
+        !ucs_list_is_empty(&self->ep_gc_list)) {
```

Comment:
i'd add assert on !ucs_list_is_empty(&self->ep_gc_list): since uct_rc_iface_cleanup_eps is really supposed to make it empty